### PR TITLE
Updated the project to use make conventions to spin up a local dockerized instance.

### PR DESCRIPTION
**Pull request by rmaheshwari-splunk**
### Update:-
    

- Enabled following **make** commands: -
1. make up - Used to spin up Splunk Docker Instance.
2. make test - Used to run test cases against the Docker instance.
3. make test_specific - Used to echo the command, used to run specific test cases against the Docker instance.

- Added docker-compose.yml to spin up Splunk instance for local and CI.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,75 +1,43 @@
...
+          java-version: 1.8
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Make Test
+        run: make test
```
Comment (zenmoto) [inline .github/workflows/test.yml:43]: Ok, this is a bit subtle, but I'd prefer to keep the make facade as just a developer aid- internally the project should use Maven, but Make should be available and consistent across projects so a developer doesn't need to know Maven in order to launch a local build.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+all: init test
+
+init:
+	@echo "$(ATTN_COLOR)==> init $(NO_COLOR)"
+
+.PHONY: test
+test:
+	@echo "$(ATTN_COLOR)==> test $(NO_COLOR)"
+	@echo "Running Unit test"
+	@mvn package -P Unittest
```
Comment (zenmoto) [inline Makefile:28]: These should be mvn verify rather than package

File: .github/workflows/test.yml (hunk 1 above)
Comment (rmaheshwari-splunk) [reply .github/workflows/test.yml]: Sure, no problem, the suggested changes have been made and have already been pushed, please take a look at them and let me know if any further changes are needed.

File: Makefile (hunk 2 above)
Comment (rmaheshwari-splunk) [reply Makefile]: Duly noted, it will reflect in the final commit along with the before mentioned change.

**Review by zenmoto — APPROVED** [review]

**Pull request merged**